src/lib/y2network/interface_config_builder.rb

```diff
@@ -0,0 +1,160 @@
+# Copyright (c) [2019] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+require "yast"
+require "y2network/interface_defaults"
+
+module Y2Network
+  # Stores what's needed when creating a new configuration for an interface
+  class InterfaceConfigBuilder
+    # @return [String] Device name (eth0, wlan0, etc.)
+    attr_accessor :name
+    # @return [String] type which is intended to be build
```

Comment:
Can you be more specific? A "type" which is a String is awfully vague.

Is network-ng using some kind of enums, where you have a clearly defined set of allowed values?

Reply:
not yet.